src/lib/rng/esdm_rng/esdm_rng.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+* ESDM RNG
+* (C) 2024, Markus Theil <theil.markus@gmail.com>
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/esdm_rng.h>
+
+#include <esdm/esdm_rpc_client.h>
+
+namespace Botan {
+
+namespace {
+class ESDM_Context {
+   private:
+      constinit inline static std::mutex g_mutex;
+
+   public:
+      [[nodiscard]] static std::shared_ptr<void> instance() {
+         std::scoped_lock lk(g_mutex);
+         static std::shared_ptr<ESDM_Context> g_instance{new ESDM_Context()};
+         return g_instance->acquire();
+      }
+
+   private:
```

Comment:
I forgot that in my godbolt example as well: Let's define the default constructor as private to make sure no-one creates another instance of this class:

```suggestion
   private:
      ESDM_Context() = default;
   
```